src/main/java/mirror/SyncDirection.java

```diff
@@ -0,0 +1,59 @@
+package mirror;
+
+public enum SyncDirection {
```

Comment:
Protobuf supports enums:

https://developers.google.com/protocol-buffers/docs/proto#enum

If we put this enum in there, then you could just send the sync direction as-is over the wire, and not have to encode it/decode it as the separate allow inbound / allow outbound properties.